core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java

```diff
@@ -89,13 +95,25 @@ private boolean isTestProjectFolder(IProject project, IResource f) {
 	 * @throws CoreException
 	 */
 	public boolean isFuseCamelContentType(IFile ifile) throws CoreException {
-		return ifile != null
-				&& ifile.isSynchronized(IResource.DEPTH_ZERO)
-				&& ifile.getContentDescription() != null
-				&& ifile.getContentDescription()
-							.getContentType()
-							.getId()
-							.equals(CamelUtils.FUSE_CAMEL_CONTENT_TYPE);
+		if( ifile != null
+				&& ifile.isSynchronized(IResource.DEPTH_ZERO)){
+			IContentDescription contentDescription = ifile.getContentDescription();
+			if(contentDescription != null){
+				String contentTypeId = contentDescription.getContentType().getId();
+				if("com.springsource.sts.config.ui.beanConfigFile".equals(contentTypeId)){
+					// Should we filter only for known conflicting content types? If not, it might impact performance.
+					try {
+						ByteArrayInputStream markableAndResettableStream = new ByteArrayInputStream(Files.readAllBytes(ifile.getLocation().toFile().toPath()));
+						return XMLContentDescriber.VALID == new CamelNamespaceXmlContentDescriber().describe(markableAndResettableStream, null);
+					} catch (IOException e) {
+						CamelModelServiceCoreActivator.pluginLog().logInfo("Cannot check Content type of "+ ifile.getName(), e);

```

Comment:
I know this is just going into the log, but should we externalize it? Or mark it to not be externalized via  //$NON-NLS-1$?